Adds Symbol levels description

SrNetoChan opened the pull request:
Fixes #908

SrNetoChan commented:
Reviews are most appreciated! @DelazJ @yjacolin thanks

SrNetoChan commented:
Any of you knows what's with the

    .. index:: Inverted_Polygons_Renderer

instead of simple references like:

    .. _inverted_polygons_renderer:

?

yjacolin commented:
@SrNetoChan this is two different things. The second is an anchor to make a reference. The first aims to create an index page (you know like a book to find where a specific word is used in the text). Not sure if I am clear but they are not currently enabled in the html page. See this direct link: http://docs.qgis.org/testing/en/genindex.html

yjacolin commented on source/docs/user_manual/working_with_vector/vector_properties.rst:
```diff
@@ -600,6 +600,55 @@ units).
+For renderers that allow stacked symbol layers (for example, 
+:ref:`single_symbol_renderer`, :ref:`inverted_polygon_renderer` and
+:ref:`rule_based_rendering`) there is an option to control the rendering order
+of each symbol level.
```
plural? level**s**

yjacolin commented on source/docs/user_manual/working_with_vector/vector_properties.rst:
```diff
@@ -600,6 +600,55 @@ units).
+
+   If symbols levels are deactivated, the complete symbols will be drawn
+   according to their respective features order. Overlapping symbols will
+   simply ofuscate to other below. Besides, similar symbols won't "merge" with 
```
obfuscate

yjacolin commented on source/docs/user_manual/working_with_vector/vector_properties.rst:
```diff
@@ -600,6 +600,55 @@ units).
    so you can edit it afterwards in the variables tab of the layer's properties dialog.
 
+.. _Symbols_levels:
+
```
could you add: `.. index:: Symbols levels` before `.. _Symbols_levels:`

yjacolin commented on source/docs/user_manual/working_with_vector/vector_properties.rst:
```diff
@@ -600,6 +600,55 @@ units).
+..............
+
+For renderers that allow stacked symbol layers (for example, 
+:ref:`single_symbol_renderer`, :ref:`inverted_polygon_renderer` and
```
`inverted_polygon_renderer` didn't exist, so I added it directly to master. Could you rebase your PR?

SrNetoChan replied:
Yes, sure.

DelazJ commented:
LGTM
However I was wondering if it would make sense to have this section near the "Draw Effects" section under the current "Other Setting" section and move them all after the "Feature Rendering" section, still under the "Style menu" section. These two options are some particular tools from the Style menu. In this case, we have:

Vector Properties
--> General menu
--> Style Menu
-----> Features rendering
-------> Single Symbol Renderer
-------> Categorized Renderer
...
...
-------> Heatmap
-------> 2.5 D
-----> Layer rendering
-----> Others Settings
-------> Symbols Levels
-------> Draw Effects
--> Labels menu
What do you think?

SrNetoChan commented:
Sounds good. I can do that in this PR as well.

DelazJ commented on source/docs/user_manual/working_with_vector/vector_properties.rst:
```diff
@@ -648,17 +648,252 @@ features of the layer:
 .. only:: html
 
-     **Figure Symbology 10:**
+   >**Figure Layer Rendering 1:**
```
typo with >

DelazJ commented on source/docs/user_manual/working_with_vector/vector_properties.rst:
```diff
@@ -648,17 +648,252 @@ features of the layer:
+For renderers that allow stacked symbol layers (for example, 
+:ref:`single_symbol_renderer`, :ref:`inverted_polygon_renderer` and
+:ref:`rule_based_rendering`) there is an option to control the rendering order
+of each symbol's levels.
```
Why not citing categorized nor graduated renderer?
Actually The "Symbols levels" widget has more to do with symbology definition than style renderer imho. Whenever the feature representation is defined using multiple levels of symbols then there's a room to use that option. And all renderers (except heatmap and 2.5D) do allow this (might be weird used with displacement point but sounds possible)

SrNetoChan commented:
Those were examples. I have choose those because at that time I did not understood the index thing. Now I can choose any other. You are right, and the image is from category renderer. Or should we mention all?

SrNetoChan commented:
Added references to all renderers sub-section for consistency, fixed the typo, updated how to access each to the symbols levels option for each renderer (rule based and point displacement are different)

DelazJ commented:
LGTM
